source/MaterialXRuntime/Private/PrvElement.cpp

```diff
@@ -62,7 +62,7 @@ void PrvCompound::addElement(PrvObjectHandle elem)
     PrvElement* el = elem->asA<PrvElement>();
     if (_elementsByName.count(el->getName()))
     {
-        throw ExceptionRuntimeError("An element named '" + el->getName().str() + "' already exists in '" + getName().str() + "'");
+        return;
```

Comment:
@niklasharrysson I fixed the runtime tests on OSX but I'm not sure it is correct.
I don't get all the logic but I can see at the [following line](https://github.com/autodesk-forks/MaterialX/blob/rt_sanbox_osx/source/MaterialXRuntime/RtCoreIo.cpp#L211) that a nodeDef could be added from a `readNode` call. In my case, it was not finding the `ND_image_float` definition and so was loading it again.

But since the test is working for you, maybe it is not the solution. Let me know what you think.

Reply:
@frenchdog Aha! Now I understand why to get this problem. So the nodedef is already loaded through readNode, if a node of that type is read before the nodedef. The ordering of elements in the MaterialX document is probably different from macOS and Windows so that is why I don't get the same problem.

I think this exception is still useful in general though, to detect name conflicts and report if a user is trying to add elements with the same name. Otherwise you will get unexpected results if you add an element that has the same name as e.g. a standard library element, and this is not reported.

So another fix would be to keep the check and error reporting here, but instead test for this particular load order problem in RtCoreIo. That is, in RtCoreIo::read check if a nodedef already exists because it has been loaded through readNode previously. I can add this fix if you like?

Reply:
Indeed, we should keep the exception in the private section and check before adding an element in RtCoreIo::read!